src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs

```diff
@@ -38,13 +39,46 @@ public abstract partial class JsonValue : JsonNode
                     return null;
                 }
 
-                if (element.ValueKind == JsonValueKind.Object || element.ValueKind == JsonValueKind.Array)
+                VerifyJsonElementIsNotArrayOrObject(ref element);
+            }
+
+            return new JsonValue<T>(value, options);
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonValue"/> class that contains the specified value.
+        /// </summary>
+        /// <returns>
+        ///   The new instance of the <see cref="JsonValue"/> class that contains the specified value.
+        /// </returns>
+        /// <typeparam name="T">The type of value to create.</typeparam>
+        /// <param name="value">The value to create.</param>
+        /// <param name="jsonTypeInfo">The <see cref="JsonTypeInfo"/> that is later used to serialize the value.</param>
+        /// <param name="options">Options to control the behavior.</param>
+        /// <returns>The new instance of the <see cref="JsonValue"/> class that contains the specified value.</returns>
+        public static JsonValue? Create<T>(T? value, JsonTypeInfo<T> jsonTypeInfo, JsonNodeOptions? options = null)
+        {
+            if (jsonTypeInfo == null)
+            {
+                throw new ArgumentNullException(nameof(jsonTypeInfo));
+            }
+
+            if (value == null)
+            {
+                return null;
+            }
+
+            if (value is JsonElement element)
+            {
+                if (element.ValueKind == JsonValueKind.Null)
                 {
-                    throw new InvalidOperationException(SR.NodeElementCannotBeObjectOrArray);
+                    return null;
                 }
+
+                VerifyJsonElementIsNotArrayOrObject(ref element);
             }
 
-            return new JsonValue<T>(value, options);
+            return new JsonValue<T>(value, jsonTypeInfo, options);
```

Comment:
Could we create a new internal type, say `JsonValueMetadataBased<T> : JsonValue` that has a different implementation of the `WriteTo` method that doesn't root calls to `options`-based `JsonSerializer` methods? On this line, we would create and return an instance of the new type instead.

This is similar to the approach taken for a size-opts friendly `JsonContent` type: https://github.com/dotnet/runtime/issues/51544. See `JsonContent<T>` [implementation](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContentOfT.cs#L38-L43). This was done so that the [new](https://github.com/dotnet/runtime/blob/e426e2ce66586bc34b618c5abbfcc9ef74d5166a/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Post.cs#L64) metadata-based System.Net.Http.Json methods do not root all converters/reflection-based code, like done in the [existing](https://github.com/dotnet/runtime/blob/e426e2ce66586bc34b618c5abbfcc9ef74d5166a/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Post.cs#L34) methods.

I think this approach could help with https://github.com/dotnet/runtime/issues/52943. From the analysis there:

> JsonNodeConverter creates new JsonValue<T>

In this model, `JsonNodeConverter` would instead create a new `JsonValueMetadataBased<T>` whose `WriteTo` /`ToString` implementation does not root existing options-based overloads.

Reply:
Can we just have `JsonValueMetadataBased<T> : JsonValue<T>` instead? This assumes the the override of `WriteTo()\ToString()` doesn't call `base.WriteTo()` or `base.ToString()` and the linker detects that.

I'll need to prototype this and verify against the object-based repro.

Reply:
> This assumes the the override of WriteTo()\ToString() doesn't call base.WriteTo() or base.ToString() and the linker detects that.

Unfortunately, this doesn't work. The linker will preserve the whole hierarchy of virtual/overrides of classes that are instantiated.

Reply:
Also I believe `ToString()` is fine based on previous conversation with @eerhardt

Reply:
> Unfortunately, this doesn't work. The linker will preserve the whole hierarchy of virtual/overrides of classes that are instantiated.

The second commit added two new classes (instead of one) so the safe and unsafe overrides are separated.